### PR TITLE
Connecting over SSL adding driver options params for mongo client in …

### PR DESCRIPTION
**Pull request by antoniojlm84**
Connecting over SSL adding driver options params for mongo client in config yaml file

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -65,6 +65,14 @@ mongo_db_bundle:
         ssl: false
         connectTimeoutMS: 3000 # default null (no timeout)
         readPreference: primaryPreferred # see https://docs.mongodb.com/manual/reference/read-preference/#primary for info
+      
+      ssl_client_name:    # see https://www.php.net/manual/en/mongo.connecting.ssl.php
```
Comment (Jean85) [inline README.MD]: This is referencing the wrong extension, the legacy one. This bundle supports `ext-mongodb`, the newer one.

File: README.MD [hunk 2]
```diff
@@ -65,6 +65,14 @@ mongo_db_bundle:
...
         connectTimeoutMS: 3000 # default null (no timeout)
         readPreference: primaryPreferred # see https://docs.mongodb.com/manual/reference/read-preference/#primary for info
+      
+      ssl_client_name:    # see https://www.php.net/manual/en/mongo.connecting.ssl.php 
+        uri: 'mongodb://host1:3062,host2'
+        password: ''
+        authSource: ''
+        ssl: true
+        driver_options:
+            context: "app.mongodb.context_service" # default null see https://symfony.com/doc/master/bundles/DoctrineMongoDBBundle/config.html#specifying-a-context-service
```
Comment (Jean85) [inline README.MD]: What's this for? Aren't we confusing stuff with a different bundle?

File: src/DependencyInjection/MongoDbBundleExtension.php [hunk 3]
```diff
@@ -93,7 +101,24 @@ private function defineConnections(array $connections): void
...
+    {
+        $driverOptions            = $clientsConfig['driver_options'] ?? [];
+        $driverOptions['typeMap'] = ['root' => 'array', 'document' => 'array'];
+        if (isset($driverOptions['context'])) {
+            $driverOptions['context'] = new Reference($driverOptions['context']);
+        }
+        return $driverOptions;
+    }
+
+    private function attachDataCollectionListenerToEventManager()
```
Comment (Jean85) [inline src/DependencyInjection/MongoDbBundleExtension.php]: You lost the return type here.

File: README.MD (hunk 2 above)
Comment (antoniojlm84) [reply README.MD]: In this section we can see an example hiow to implement a service which creates the correct context stream. But I can remove it and add an example in the docs where you thinks is the right place.

Comment (antoniojlm84) [reply README.MD]: I have looked at the offical doctrine documentation on at symfony.com to try to do things as symfony way as possible. So, I build a solution for the driver options like doctrine does.

Service example:
app.mongodb.context_service:
        class: 'resource'
        factory: 'stream_context_create'
        arguments:
            - { ssl: { verify_expiry: true } }

File: src/DependencyInjection/MongoDbBundleExtension.php (hunk 3 above)
Comment (antoniojlm84) [reply src/DependencyInjection/MongoDbBundleExtension.php]: you are right, it should be:

* @return array|mixed

**Pull request closed without merging**